synapse/handlers/room_summary.py

```diff
@@ -814,7 +821,10 @@ async def _is_local_room_accessible(
             logger.info("room %s is unknown, omitting from summary", room_id)
             return False
 
-        room_version = await self._store.get_room_version(room_id)
+        try:
+            room_version = await self._store.get_room_version(room_id)
+        except UnsupportedRoomVersionError:
+            return False
```

Comment:
Can you add a bit of a comment as to why this is OK, ie "this stops us breaking the summary in the case where we drop room version support", or something?